modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -3819,16 +3819,19 @@ private void validatePartitionsState() {
 
                     // Do not validate read or write through caches or caches with disabled rebalance
                     // or ExpiryPolicy is set or validation is disabled.
-                    boolean eternalExpiryPolicy = grpCtx != null && (grpCtx.config().getExpiryPolicyFactory() == null
-                        || grpCtx.config().getExpiryPolicyFactory().create() instanceof EternalExpiryPolicy);
+                    boolean customExpiryPolicy = Optional.ofNullable(grpCtx)
+                        .map((v) -> v.caches())
```

Comment:
Please use java 8 style -> CacheGroupContext::caches, Collections::emptyList

Reply:
To be honest, I don't see any advantages to use method reference instead of lambdas here. It is a matter of taste. Anyway, I will change it if you want.
By the way, please take a look at this comment from Brian Goetz: https://stackoverflow.com/a/24493905/8371000